### PR TITLE
[Enhancement] update csv reader to properly parse comma wrapped in quotes

### PR DESCRIPTION
**Pull request by HelloKitty1024**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: be/src/formats/csv/csv_reader.cpp [hunk 1]
```diff
@@ -1,7 +1,6 @@
 // This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
 
 #include "formats/csv/csv_reader.h"
-
 namespace starrocks::vectorized {
 
 Status CSVReader::next_record(Record* record) {
```
[the remark this answers is not in the document]
Comment (normanyahq) [reply be/src/formats/csv/csv_reader.cpp:6]: @zaorangyang in order to handle that, a larger refactor is required - basically line reader and field reader should be done together. The current implementation doesn't handle this case either, this PR addresses the one common case without introducing too much overhead.

[the remark this answers is not in the document]
Comment (HelloKitty1024) [reply be/src/formats/csv/csv_reader.cpp:6]: @zaorangyang in order to handle that, a larger change is required - basically line reader and field reader should be combined together. The current implementation doesn't handle this case either, and this PR addresses the one common case without introducing too much overhead.

[the remark this answers is not in the document]
Comment (zaorangyang) [reply be/src/formats/csv/csv_reader.cpp:6]: In my opinion this case may happen frequently, so it must be solved.

[the remark this answers is not in the document]
Comment (HelloKitty1024) [reply be/src/formats/csv/csv_reader.cpp:6]: @zaorangyang yeah sure, I can fix it with the next PR. Can you review and merge this first as it's not worse than the status quo?

[the remark this answers is not in the document]
Comment (jaogoy) [reply be/src/formats/csv/csv_reader.cpp:6]: @HelloKitty1024 do you have any design plan to handle this? if not, such an uncompleted improvement will be release to users, while common cases of using quote are to enclose field separator and line delimiter.

**Comment by HelloKitty1024** [comment]
closing it as https://github.com/StarRocks/starrocks/pull/13582 is a more comprehensive fix

**Pull request closed without merging**